circom-prover/src/prover/arkworks.rs

```diff
@@ -58,23 +59,21 @@ pub fn generate_circom_proof(
     Ok(CircomProof { proof, pub_inputs })
 }
 
-pub fn verify_circom_proof(
-    zkey_path: String,
-    proof: Vec<u8>,
-    public_inputs: PublicInputs,
-) -> Result<bool> {
+pub fn verify_circom_proof(zkey_path: String, proof: CircomProof) -> Result<bool> {
     let mut header_reader = ZkeyHeaderReader::new(&zkey_path);
     header_reader.read();
     let file = File::open(&zkey_path)?;
     let mut reader = std::io::BufReader::new(file);
     if header_reader.r == BigUint::from(ark_bn254::Fr::MODULUS) {
         let proving_key = read_proving_key::<_, Bn254>(&mut reader)?;
-        let serialized_inputs: SerializableInputs<Bn254> = public_inputs.into();
-        verify(proving_key.vk, serialized_inputs.0, proof)
+        let serialized_proof =
+            serialization::serialize_proof::<Bn254>(&SerializableProof(proof.proof.into()));
```

Comment:
I think we don't need to convert to `SerializableProof`.
Hopefully we just need to convert between `CircomProof` and `use ark_groth16::Proof;`

Reply:
Good idea! fixed in [a0551c9](https://github.com/zkmopro/mopro/pull/372/commits/a0551c983b38309cd18d6e21617883a74b541b1f)